### PR TITLE
Gemify chefdk-julia

### LINKED ISSUE #1: Add files needed to turn chefdk-julia into a gem
Need a gemspec, supporting files (version, license, etc)

Also update README and add a path method for finding the location of the cookbook.

Add a post-installation about configuring ChefDK to use Julia.

### PR DESCRIPTION
**Pull request by dougireton**
Add gemspec
Add Rake gem tasks
Add Apache 2.0 license

Fixes #1

### CODE REVIEW

**Comment by dougireton** [comment]
@oifland can you review?

**Comment by oifland** [comment]
I'd be happy to, @dougireton. Can I get collaborator status so I can merge when I'm done?

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (oifland) [inline LICENSE]: This needs to be filled in

File: README.md [hunk 2]
```diff
@@ -1,38 +1,39 @@
 # chefdk-julia
-An Opinionated Cookbook Creator
+An Opinionated Cookbook creator.
```
Comment (oifland) [inline README.md]: If you're turning this into a sentence, we should drop the capital letters consistently (it was a subtitle previously)

File: README.md [hunk 3]
```diff
@@ -1,38 +1,39 @@
...
-Julia is a special type of Chef cookbook that is designed to generate code
-(cookbooks, templates, etc).
+Julia is a special type of Chef cookbook that is designed to generate code (cookbooks, recipes, templates, etc.).
 
-It is based upon the default code_generator cookbook introduced in ChefDK 0.9.0
+It is based upon the default code_generator provided by ChefDK.
 (`chef generate generator`).
 
+## Prerequisites
+[ChefDK](https://chef.io/downloads) must already be installed.
```
Comment (oifland) [inline README.md]: Change URL to https://downloads.chef.io/chef-dk/

File: README.md [hunk 4]
```diff
@@ -1,38 +1,39 @@
...
+
 ## Installation
-For now, we're recommending grabbing the latest version from git:
+`chefdk-julia` is distributed as a Ruby Gem.
 
-    cd ~/chef/cookbooks
-    git clone ssh://git@git.nordstrom.net/chf/chefdk-julia.git
+Install it with `chef gem install` on the command line.
```
Comment (oifland) [inline README.md]: `chef gem install chefdk-julia` perhaps?

File: README.md [hunk 5]
```diff
@@ -1,38 +1,39 @@
...
 
-    chefdk.generator_cookbook "~/chef/cookbooks/chefdk-julia"
+Add these lines to your `~/.chef/config.rb` or `~/.chef/knife.rb` configuration file:
 
-Adjust the `~/chef/cookbooks/chefdk-julia` path as needed if you put it somewhere
-else. Be sure that the folder's name is 'chefdk-julia' so that it matches the
-metadata.rb's name for the cookbook.
+```ruby
+# Paste lines below into ~/.chef/config.rb or knife.rb
+require 'chefdk-julia'
```
Comment (oifland) [inline README.md]: Do we need that "if" branch around it?

File: lib/chefdk/julia/version.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module Chefdk
```
Comment (oifland) [inline lib/chefdk/julia/version.rb]: Headers?

File: chefdk-julia.gemspec [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# coding: utf-8
```
Comment (oifland) [inline chefdk-julia.gemspec]: Headers?

File: README.md [hunk 8]
```diff
@@ -1,38 +1,39 @@
...
-the `chef` command:
+Once chefdk-julia is installed, you should be able to generate a cookbook with the `chef generate cookbook` command:
 
-    chef generate cookbook the_art_of_french_cooking
+```bash
+> chef generate cookbook the_art_of_french_cooking
+```
 
-where `the_art_of_french_cooking` is the name of the cookbook you want to create.
+where '*the_art_of_french_cooking*' is the name of the cookbook you want to create.
```
Comment (oifland) [inline README.md]: Should we mention the license in the readme?

File: LICENSE (hunk 1 above)
Comment (dougireton) [inline LICENSE]: I think in this case it's fine since it's the just the generic instructions on how to apply the license.

File: README.md (hunk 2 above)
Comment (dougireton) [inline README.md]: agreed

File: README.md (hunk 3 above)
Comment (dougireton) [inline README.md]: meant to do that. Thanks!

File: README.md (hunk 5 above)
Comment (dougireton) [inline README.md]: Makes it more complicated, but is more correct and protects the user if they don't have ChefDK installed. Do you think that guard is worth the extra complexity?

File: lib/chefdk/julia/version.rb (hunk 6 above)
Comment (dougireton) [inline lib/chefdk/julia/version.rb]: Yes. Missed that one.

File: chefdk-julia.gemspec (hunk 7 above)
Comment (dougireton) [inline chefdk-julia.gemspec]: D'oh!

File: README.md (hunk 8 above)
Comment (dougireton) [inline README.md]: Absolutely

**Pull request merged**